src/test/status_test.cpp

```diff
@@ -36,4 +39,20 @@ TEST(StatusCodeTest, AllStatusCodesMapped) {
         ASSERT_NE(status.string(), "Undefined error");
     }
 }
+
+TEST(StatusCodeTest, CApi) {
+    for (auto statusCode = StatusCode::OK; statusCode != StatusCode::STATUS_CODE_END; ++statusCode) {
+        Status status = Status(statusCode);
```

Comment:
You are using here copy constructor in L46. Please check what happens if you try to get details from status object from L45

Reply:
why not use &status with reinterpret_cast? Previous comment is invalid

Reply:
done